perl/bin/ascatCounts.pl

```diff
@@ -0,0 +1,217 @@
+  Sanger::CGP::Ascat::Implement::prepare($options);
+  my $threads = PCAP::Threaded->new($options->{'threads'});
+  
+  # register any process that can run in parallel here
+  $threads->add_function('allele_count', \&Sanger::CGP::Ascat::Implement::allele_count);
+
+  # start processes here (in correct order obviously), add conditions for skipping based on 'process' option
+  my $jobs = $options->{'lociChrsBySample'};
+  $jobs = $options->{'limit'} if(exists $options->{'limit'} && defined $options->{'limit'});
+  $threads->run($jobs, 'allele_count', $options);
+
+  merge_allele_counts($options);
+  cleanup($options) unless($options->{'noclean'} == 1);
+}
+
+sub cleanup {
+  my $options = shift;
+  my $tmpdir = $options->{'tmp'};
+  move(File::Spec->catdir($tmpdir, 'ascatCounts'), File::Spec->catdir($options->{'outdir'}, 'ascatCounts')) || die $!;
+  move(File::Spec->catdir($tmpdir, 'logs'), File::Spec->catdir($options->{'outdir'}, 'logs')) || die $!;
+  remove_tree $tmpdir if(-e $tmpdir);
+  return 0;
+}
+
+sub _which {
+  my $prog = shift;
+  my $l_bin = $Bin;
+  my $path = File::Spec->catfile($l_bin, $prog);
+  $path = which($prog) unless(-e $path);
+  die "Failed to find $prog in PATH or local bin folder" unless(defined $path);
+  return $path;
+}
+
+sub setup {
+  my %opts;
+  pod2usage(-msg  => "\nERROR: Option must be defined.\n", -verbose => 1,  -output => \*STDERR) if(scalar @ARGV == 0);
+  $opts{'cmd'} = join " ", $0, @ARGV;
+  GetOptions( 'h|help' => \$opts{'h'},
+              'm|man' => \$opts{'m'},
+              'v|version' => \$opts{'v'},
+              'o|outdir=s' => \$opts{'outdir'},
+              'b|bam=s' => \$opts{'bam'},
+              'c|cpus=i' => \$opts{'threads'},
+              'q|minbasequal=i' => \$opts{'minbasequal'},
+              'sg|snp_gc=s' => \$opts{'snp_gc'},
+              'r|reference=s' => \$opts{'reference'},
+              'nc|noclean' => \$opts{'noclean'},
+  ) or pod2usage(2);
+
+  pod2usage(-verbose => 1, -exitval => 0) if(defined $opts{'h'});
+  pod2usage(-verbose => 2, -exitval => 0) if(defined $opts{'m'});
+
+  if($opts{'v'}){
+    print Sanger::CGP::Ascat->VERSION."\n";
+    exit;
+  }
+
+  warn "Executing: $opts{cmd}\n";
+
+  if(!defined($opts{'noclean'})){
+    $opts{'noclean'} = 0;
+  }
+
+  # then check for no args:
+  my $defined;
+  for(keys %opts) { $defined++ if(defined $opts{$_}); }
+  pod2usage(-msg  => "\nERROR: Options must be defined.\n", -verbose => 1,  -output => \*STDERR) unless($defined);
+
+  for my $item (qw(bam snp_gc reference outdir)) {
+    pod2usage(-msg  => "\nERROR: Option '-$item' must be defined.\n", -verbose => 1,  -output => \*STDERR) unless(defined $opts{$item});
+  }
+
+  for my $item(qw(bam reference outdir)) {
+    $opts{$item} = File::Spec->rel2abs( $opts{$item} ) if(defined $opts{$item});
+  }
+
+  $opts{'tumour'} = $opts{'bam'}; #need to map bam to tumour to use existing perl modules
+  $opts{'normal'} = $opts{'tumour'}; #need to define normal too
```

Comment:
This isn't necessary if you move the name populate into here as described against the call to `prepare()`, see earlier comment